src/layout/mod.rs

```diff
@@ -5174,6 +5263,80 @@ mod tests {
                 }
                 Op::MoveWorkspaceDown => layout.move_workspace_down(),
                 Op::MoveWorkspaceUp => layout.move_workspace_up(),
+                Op::MoveWorkspaceToIndex {
+                    ws_name: Some(ws_name),
+                    target_idx,
+                } => {
+                    let MonitorSet::Normal { monitors, .. } = &mut layout.monitor_set else {
+                        return;
+                    };
+
+                    let (old_idx, old_output) = monitors
+                        .iter()
+                        .find_map(|monitor| {
+                            monitor
+                                .workspaces
+                                .iter()
+                                .enumerate()
+                                .find_map(|(i, ws)| {
+                                    if ws.name == Some(format!("ws{ws_name}")) {
+                                        Some(i)
+                                    } else {
+                                        None
+                                    }
+                                })
+                                .map(|i| (i, monitor.output.clone()))
+                        })
+                        .unwrap();
```

Comment:
Doesn't this fail when the workspace doesn't exist? These are randomized tests so they shouldn't panic if the randomly generated ws name doesn't exist.

Reply:
Also don't forget to actually run these tests: https://github.com/YaLTeR/niri/wiki/Developing-niri#running-tests